pkg/trace/agent/agent.go

```diff
@@ -234,25 +233,11 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 
 		events, keep := a.sample(ts, pt)
 
-		if sublayerCalculator.ShouldCompute(keep) {
-			pt.Sublayers = make(map[*pb.Span][]stats.SublayerValue)
-			subtraces := stats.ExtractSubtraces(t, root)
-			for _, subtrace := range subtraces {
-				subtraceSublayers := sublayerCalculator.ComputeSublayers(subtrace.Trace)
-				if sublayerCalculator.WithStats() {
-					pt.Sublayers[subtrace.Root] = subtraceSublayers
-				}
-				if keep {
-					stats.SetSublayersOnSpan(subtrace.Root, subtraceSublayers)
-				}
-			}
-		}
 		sinputs = append(sinputs, stats.Input{
-			Trace:         pt.WeightedTrace,
-			Sublayers:     pt.Sublayers,
-			Env:           pt.Env,
-			SublayersOnly: p.ClientComputedStats,
+			Trace: pt.WeightedTrace,
```

Comment:
if ClientComputedStats is set, you can actually stop filling sinputs and not allocate it